domain/operation/doc.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package operation defines the domain model, contracts, and documentation for
+// Juju operations.
+//
+// An operation represents a higher-level unit of work initiated by a user or
+// by Juju itself, typically comprising one or more actions/tasks executed
+// against applications, units, or machines. The operation domain provides the
+// types shared across the service and state subpackages, and serves as the
+// package-level documentation entrypoint for the operation domain.
+//
+// ## Package layout
+//   - service: high-level business API used by apiserver facades and workers to
+//     manage operations for a model. It is intentionally decoupled from the
+//     state layer.
+//   - state: access to persistence (controller/model database) relating to
+//     operations. It exposes a minimal interface consumed by the service.
+//   - types.go: DTOs used as arguments/results for state and service calls where
+//     they need to be visible to both subpackages.
```

Comment:
This is not exclusive to this package. I'd prefer not having to document this anywhere, since the package layout (service/state/errors/etc) is pretty standard and self explanatory, but we decide to do it then it should be on the root domain/ package.